packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -292,34 +308,16 @@ export function attemptToDispatchEvent(
           // priority for this boundary.
           return instance;
         }
-        // This shouldn't happen, something went wrong but to avoid blocking
-        // the whole system, dispatch the event without a target.
-        // TODO: Warn.
-        targetInst = null;
       } else if (tag === HostRoot) {
         const root: FiberRoot = nearestMounted.stateNode;
         if (root.isDehydrated) {
           // If this happens during a replay something went wrong and it might block
           // the whole system.
           return getContainerFromFiber(nearestMounted);
```

Comment:
So before this function did two things:
1. Find the instance blocking this event
2. Dispatch the event if nothing is blocking it.

I broke up the functionality so it only does (1) now and outside of this function we call `dispatchEventForPluginEventSystem`  only if there is nothing blocking the instance. The reason for this is because for replay I opted not to use `dispatchEventForPluginEventSystem` because this would only play a particular phase; I could have saved 2 events, 1 for capture and 1 for bubble phase of the event and called `dispatchEventForPluginEventSystem` twice,  Instead I opted for redispatching the event natively that way we wouldn't need to add extra logic to hook into `stopPropagation` and stop the bubble phase. This means we'll re-enter ReactDOMEventListener but this time around when we see nothing is blocking the event we'll call `dispatchEventForPluginEventSystem`. If the event has stopPropagation called on it in the capture phase then we'll never enter ReactDOMEventListener for the bubble phase so that'll work out of the box without needing to hook into anything.

Reply:
Sorry, I am specifically referring to this codepath:

```js
        if (root.isDehydrated) {
          // If this happens during a replay something went wrong and it might block
          // the whole system.
          return getContainerFromFiber(nearestMounted);
```

Previously, before this PR, this early return would've caused `dispatchEventForPluginEventSystem` to **not** be called for this particular codepath. With this PR, `dispatchEventForPluginEventSystem` is called in this codepath.

I'm not against separating the two things, but this particular codepath, from my observation, has changed behavior.

Reply:
It shouldn't run because it's returning the `container` so `blockedOn` is non null so we won't dispatch. Let me double check thats the case though